node/sync.go

```diff
@@ -476,7 +519,19 @@ func (d *Pegnetd) applyTransactionBatch(sqlTx *sql.Tx, txBatch *fat2.Transaction
 			return err
 		}
 
-		if tx.IsConversion() {
+		// All conversions to PEG after the activation height have their
+		// outputs processed later. We only subtract their inputs right now.
+		if currentHeight >= PegnetConversionLimitActivation && tx.IsPEGRequest() {
+			// Ensure the output is valid, as we will process it later
+			_, err := conversions.Convert(int64(tx.Input.Amount), rates[tx.Input.Type], rates[tx.Conversion])
+			if err != nil {
+				return err
+			}
+			return nil // PEG Outputs are handled elsewhere
```

Comment:
I think this should be a "continue", not a "return nil". Otherwise if you have a txbatch with a X => PEG conversion and a X => Y conversion, the second conversion would never get recorded

Reply:
Good catch.